static/js/entry/zendesk_widget.js

```diff
@@ -116,6 +118,46 @@ const zendeskCallbacks = {
         }, 100);
       };
     }
+  },
+  ticketSubmissionFormLoaded: () => {
+    const iframe = document.querySelector("iframe.zEWidget-ticketSubmissionForm");
+
+    const fieldSelector = name => (
+      `input[name="${name}"], select[name="${name}"]`
+    );
+    const iframeQuerySelector = query => (
+      iframe.contentDocument.querySelector(query)
+    );
+    const fieldElement = R.compose(
+      iframeQuerySelector, fieldSelector
+    );
+
+    // Zendesk uses the ID 24690866 to refer to the MicroMasters program selector
+    const programFieldName = '24690866';
+    const fieldVisibility = {
+      'name':  !(SETTINGS.user && SETTINGS.user.first_name
+                 && SETTINGS.user.last_name),
+      'email': !(SETTINGS.user && SETTINGS.user.email),
+    };
+    fieldVisibility[programFieldName] = !(
+      SETTINGS.program && SETTINGS.program.slug
+    );
```

Comment:
Yes, this is a bug that bit me while I was developing this feature. Let's say you have this code:
```python
variable = "value"
obj = {
   "literal": "foo",
   variable: "bar",
}
```
If you run that in Python, you'll end up with an object that looks like this:
```python
{"literal": "foo", "value": "bar"}
```
But if you run that in JavaScript, you'll get this:
```js
{"literal": "foo", "variable": "bar"}
```

I had to do this as a workaround instead. I wish there was a way to do this properly in Javascript, specifying that you want it to evaluate the variable for the key.